lib/ansible/modules/network/vyos/vyos_linkagg.py

```diff
@@ -0,0 +1,242 @@
+"""
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vyos import load_config, run_commands
+from ansible.module_utils.vyos import vyos_argument_spec, check_args
+
+
+def search_obj_in_list(name, lst):
+    for o in lst:
+        if o['name'] == name:
+            return o
+
+    return None
+
+
+def map_obj_to_commands(updates, module):
+    commands = list()
+    want, have = updates
+    state = module.params['state']
+
+    for w in want:
+        name = w['name']
+        members = w.get('members') or []
+        mode = w['mode']
+
+        if mode == 'on':
+            mode = '802.3ad'
+
+        state = w['state']
+
+        obj_in_have = search_obj_in_list(name, have)
+
+        if state == 'absent':
+            if obj_in_have:
+                for m in obj_in_have['members']:
+                    commands.append('delete interfaces ethernet ' + m + ' bond-group')
+
+                commands.append('delete interfaces bonding ' + name)
+        else:
+            if not obj_in_have:
+                commands.append('set interfaces bonding ' + name + ' mode ' + mode)
+
+                for m in members:
+                    commands.append('set interfaces ethernet ' + m + ' bond-group ' + name)
+
+                if state == 'down':
+                    commands.append('set interfaces bonding ' + name + ' disable')
+            else:
+                if mode != obj_in_have['mode']:
+                    commands.append('set interfaces bonding ' + name + ' mode ' + mode)
+
+                missing_members = list(set(members) - set(obj_in_have['members']))
+                for m in missing_members:
+                    commands.append('set interfaces ethernet ' + m + ' bond-group ' + name)
+
+                if state == 'down' and obj_in_have['state'] == 'up':
+                    commands.append('set interfaces bonding ' + name + ' disable')
+                elif state == 'up' and obj_in_have['state'] == 'down':
+                    commands.append('delete interfaces bonding ' + name + ' disable')
+
+    return commands
+
+
+def map_config_to_obj(module):
+    obj = []
+    output = run_commands(module, ['show interfaces bonding slaves'])
+    lines = output[0].splitlines()
+
+    if len(lines) > 1:
+        for line in lines[1:]:
+            splitted_line = line.split()
+
+            name = splitted_line[0]
+            mode = splitted_line[1]
+            state = splitted_line[2]
+
+            if len(splitted_line) > 4:
+                members = splitted_line[4:]
+            else:
+                members = []
+
+            obj.append({'name': name,
+                        'mode': mode,
+                        'members': members,
+                        'state': state})
+
+    return obj
+
+
+def map_params_to_obj(module):
+    obj = []
+
+    if 'collection' in module.params and module.params['collection']:
+        for c in module.params['collection']:
+            d = c.copy()
```

Comment:
Is it better to validate the parameters in the collection?
 Logic [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/junos/junos_user.py#L183-L186) can be used as a reference.